### ISSUE TITLE
Creating type from existing GraphQLInputObjectType

### ISSUE THREAD

**Issue by synthecypher** [graphql-compose/graphql-compose#176]
I'm trying to add a new field which resolved related models but also provides `filter` but with the field that is used to related the two models removed from the type.

Essentially what was suggested in this [post](https://github.com/graphql-compose/graphql-compose/issues/94#issuecomment-334407558).

```
const findManyResolver = lotTypeComposer.getResolver('findMany');
tc.addFields({
  lots: {
    type: findManyResolver,
    args: findManyResolver.getArgs(),
    resolve: (source, args, context, info) =>
        findManyResolver.resolve({
            source,
            Object.assign({}, args, {
                filter: {
                    _id: source.relatedId
                }
            }),
            context,
            info
        })
    }
  }
});
```

So essentially I want to create a copy of `findManyResolver.getArgs()` where the `filter` property has a new name and the `_id` field removed.

**Comment by nodkz** [comment]
Try to use `findManyResolver.getArgTC('filter').clone('MyNewTypeNameForFilter')`. Modify this clonned type as you need and set it as a new arg type.

**Comment by synthecypher** [comment]
@nodkz thanks for the reply I've done that but it won't effect all the other related fields like the `OR` and `AND` fields so I'll have to clone and update all of those as well.

Is there not another easier way?

It would be quite nice to specify the related keys as a option of `addRelation` and have it take care of removing that key while keeping the `filter`.

**Comment by synthecypher** [comment]
I got something I'm satisfied with sharing for future Googlers.
```

const { getOrderTC } = require('./order');
const orderTC = getOrderTC();
const orderFindManyResolver = orderTC.getResolver('findMany').clone();
const orderFilterArg = orderFindManyResolver.getArgTC('filter').clone('FilterFindManyCustomerOrderInput');
orderFilterArg.removeField('customerId');
orderFilterArg.setField('OR', orderFilterArg);
orderFilterArg.setField('AND', orderFilterArg);
orderFindManyResolver.setArg('filter', orderFilterArg);

CustomerTC.addFields({
    orders: () => ({
        type: orderFindManyResolver,
        args: orderFindManyResolver.getArgs(),
        resolve: (source, args, context, info) =>
            orderFindManyResolver.resolve({
                source,
                args: Object.assign({}, args, {
                    filter: Object.assign({}, args.filter, {
                        customerId: source._id
                    })
                }),
                context,
                info
            }),
        projection: {
            _id: true
        }
    })
});
```

**Comment by nodkz** [comment]
Yeah, `graphql-compose` does not have a helper for cloning inner types.

BUT `graphql-compose` designed in such way where you may write your own function generators/modifiers which will prepare types by your rules.

Just write your function which accepts config options and by these options call `clone`, `add*`, `remove*`, `extend*` methods as you need. You'll save a lot of time with this approach when needing complex type generation which requires many copy-paste operations by hands. Delegate it to your own helper functions.

In such way works `graphql-compose-mongoose`. It takes mongoose models and under the hood call `graphql-compose` methods for types creation.

**Comment by nodkz** [comment]
Just write your own resolve method 😈
See how implemented resolvers here https://github.com/graphql-compose/graphql-compose-mongoose/tree/master/src/resolvers

You became an advanced user, and must move to one level upper 👍

**Comment by nodkz** [comment]
`graphql-compose-mongoose` generates resolvers, but you may write your own. They are very simple.

PS. Don't forget to contribute to `graphql-compose-mongoose` when write something cool 😜
